sdk/storage/azure-storage-blob-batch/README.md

```diff
@@ -0,0 +1,203 @@
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-core-http-okhttp</artifactId>
+  <version>1.0.0-preview.5</version>
+</dependency>
+```
+
+### Configuring HTTP Clients
+When an HTTP client is included on the classpath, as shown above, it is not necessary to specify it in the client library [builders](#create-blobserviceclient), unless you want to customize the HTTP client in some fashion. If this is desired, the `httpClient` builder method is often available to achieve just this, by allowing users to provide a custom (or customized) `com.azure.core.http.HttpClient` instances.
+
+For starters, by having the Netty or OkHTTP dependencies on your classpath, as shown above, you can create new instances of these `HttpClient` types using their builder APIs. For example, here is how you would create a Netty HttpClient instance:
+
+```java
+HttpClient client = new NettyAsyncHttpClientBuilder()
+    .port(8080)
+    .wiretap(true)
+    .build();
+```
+
+### Create a Storage Account
+To create a Storage Account you can use the Azure Portal or [Azure CLI][storage_account_create_cli].
+
+```Powershell
+az stoage account create \
+    --resource-group <resource-group-name> \
+    --name <storage-account-name> \
+    --location <location>
+```
+
+## Key concepts
+
+Blob storage is designed for:
+
+- Serving images or documents directly to a browser.
+- Storing files for distributed access.
+- Streaming video and audio.
+- Writing to log files.
+- Storing data for backup and restore, disaster recovery, and archiving.
+- Storing data for analysis by an on-premises or Azure-hosted service.
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Azure Storage Blob Batch tasks, including:
+
+- [Creating BlobBatchClient](#create-blobbatchclient)
+- [Bulk Deleting Blobs](#bulk-deleting-blobs)
+- [Bulk Setting AccessTier](#bulk-setting-accesstier)
+- [Advanced Batching](#advanced-batching)
+
+### Creating BlobBatchClient
+
+Create a BlobBatchClient from a [`BlobServiceClient`]().
+```java
+BlobBatchClient blobBatchClient = new BlobBatchClientBuilder(blobServiceClient).buildClient();
+```
+
+### Bulk Deleting Blobs
+
+```java
+blobBatchClient.deleteBlobs(blobUrls, DeleteSnapshotsOptionType.INCLUDE).forEach(response ->
+    System.out.printf("Deleting blob with URL %s completed with status code %d%n",
+        response.getRequest().getUrl(), response.getStatusCode()));
+```
+
+### Bulk Setting AccessTier
+
+```java
+blobBatchClient.setBlobsAccessTier(blobUrls, AccessTier.HOT).forEach(response ->
+    System.out.printf("Setting blob access tier with URL %s completed with status code %d%n",
+        response.getRequest().getUrl(), response.getStatusCode()));
+```
+
+### Advanced Batching
+
+Deleting blobs in a batch that have different pre-requisites.
+```java
+BlobBatch blobBatch = blobBatchClient.getBlobBatch();
+
+// Delete a blob.
+Response<Void> deleteResponse = blobBatch.deleteBlob(blobUrl);
```

Comment:
should this be deleteBlobWithResponse?
Also, the methods below.

Reply:
This is as it should be given how batching is implemented.